tests/unit/dbapi_test.py

```diff
@@ -0,0 +1,133 @@
+import unittest
+
+from hazelcast.config import Config
+from hazelcast.db import _make_config, InterfaceError
+
+
+class DbApiTestCase(unittest.TestCase):
+    def test_make_config_invalid(self):
+        test_cases = [
+            (Config(), {"dsn": "hz://"}),
+            (Config(), {"user": "some-user"}),
+            (None, {"dsn": "hz://", "password": "some-pass"}),
+        ]
+        for c, kwargs in test_cases:
+            self.assertRaises(InterfaceError, lambda: _make_config(c, **kwargs))
+
+    def test_make_config_default(self):
+        cfg = _make_config()
+        target = config_with_values(cluster_members=["localhost:5701"])
+        self.assertEqualConfig(target, cfg)
+
+    def test_make_config_config_object(self):
+        cfg = config_with_values(creds_username="joe")
+        target = config_with_values(creds_username="joe")
+        self.assertEqualConfig(target, cfg)
+
+    def test_make_config_cluster(self):
+        cfg = _make_config(cluster_name="foo")
+        target = config_with_values(
+            cluster_members=["localhost:5701"],
+            cluster_name="foo",
+        )
+        self.assertEqualConfig(target, cfg)
+
+    def test_make_config_user_password(self):
+        cfg = _make_config(user="joe", password="jane")
+        target = config_with_values(
+            cluster_members=["localhost:5701"],
+            creds_username="joe",
+            creds_password="jane",
+        )
+        self.assertEqualConfig(target, cfg)
+
+    def test_make_config_host(self):
+        cfg = _make_config(host="foo.com")
+        target = config_with_values(cluster_members=["foo.com:5701"])
+        self.assertEqualConfig(target, cfg)
+
+    def test_make_config_port(self):
+        cfg = _make_config(port=1234)
+        target = config_with_values(cluster_members=["localhost:1234"])
+        self.assertEqualConfig(target, cfg)
+
+    def test_make_config_host_port(self):
+        cfg = _make_config(host="foo.com", port=1234)
+        target = config_with_values(cluster_members=["foo.com:1234"])
+        self.assertEqualConfig(target, cfg)
+
+    def test_make_config_dsn(self):
+        test_cases = [
```

Comment:
we are using parameterized tests in the python client for such things

Reply:
I think the current approach is simpler and easier to understand.

Reply:
Maybe simpler to write but when things go wrong, they are not good.
- We can't run all the test cases once the first one fails
- We can't know which test case failed directly from the output

Reply:
The current way of implementing this thing in the code base looks clunky and very limited to me. Using subtests would be better, but it requires a depencency for pytest: https://pypi.org/project/pytest-subtests/ 

> We can't run all the test cases once the first one fails

Is it that important? It takes a second to run the tests again if one of them fails.

> We can't know which test case failed directly from the output

I've improved the tests to display a message about the test case at: https://github.com/hazelcast/hazelcast-python-client/pull/586/commits/41a3be8afdc5bf39cdeac88cc0a34fdc3210be66